lms/djangoapps/learning_success/management/commands/export_all_activity_records.py

```diff
@@ -304,32 +304,44 @@ class Command(BaseCommand):
 
     def add_arguments(self, parser):
         parser.add_argument('source_platform', type=str)
-        parser.add_argument('program_code', type=str)
+        parser.add_argument('programme_id', type=str)
 
-    def handle(self, source_platform, program_code, **kwargs):
+    def handle(self, source_platform, programme_id, **kwargs):
         """POST the collected data to the api endpoint from the settings
             Arguments:
                 source_platform: Platform import as, i.e. 'juniper' or 'ginkgo'
                 program_code: Program code of program to use 'disd'
         """
 
-        program = get_program_by_program_code(program_code)
-        student_data = list(all_student_data(program))
+        programme = get_program_by_program_code(programme_id)
+        student_data = list(all_student_data(programme))
 
         engine = create_engine(CONNECTION_STRING, echo=False)
         with engine.begin() as conn:
-            # remove existing
-            conn.execute("delete from lms_activity where source_platform = %s", (source_platform,))
-
+            # remove any existing rows for the day if exists
+            conn.execute(
+                ("DELETE FROM lms_records WHERE source_platform = %s "
+                 "AND programme_id = %s "
+                 "AND DATE(created) = %s;"), (
+                    source_platform, programme_id,
+                    datetime.now().strftime(r'%Y-%m-%d')))
+
+            # format data such that all the student_data is in one cell
+            formatted_student_data = [
+                {'email': student.get('email'), 
+                 'student_data': json.dumps(student, default=str)}
+                for student in student_data]
             # add new
-            df = pd.DataFrame(student_data)
+            df = pd.DataFrame(formatted_student_data)
             df['created'] = datetime.now()
             # TODO: Add arg for source_platform
             df['source_platform'] = source_platform
-            df['program_code'] = program_code
+            df['programme_id'] = programme_id
+            df['state'] = 'initial'
             write_type = 'append'
```

Comment:
was there any particular reason not to inline it into line 346? Was this intended to be an arg?

Reply:
No particular reason, I think there was a condition there before. Changed to set it inline.